fix(flow_runner): language fallback logic

joaothallis opened the pull request:
### Issue Description

The current implementation of the `FlowRunner.Spec.Resource` module has a limitation in handling language fallback language (when the `context.language` language is not in the list of `flow` languages), it's returning an error `{:error, "no matching languages for resource"}`.

### Solution

The proposed changes address this issue by using default language when `context.language` not in `flow` languages.

joaothallis committed:
fix(flow_runner): language fallback logic

joaothallis committed:
refactor(resources): change variable for `resources`

joaothallis commented:
I have read the CLA Document and I hereby sign the CLA

joaothallis committed:
test(resource): using default language

santiagocardo commented on lib/flow_runner/spec/resource.ex:
```diff
@@ -24,27 +24,26 @@ defmodule FlowRunner.Spec.Resource do
+        %FlowRunner.Context{language: language}
+      )
+
+    # Filter resources by language and mode.
+    matching_source? = fn resource ->
+      resource.language_id == language.id && ResourceValue.supports_mode(resource, mode)
+    end
+
+    resource_values = Enum.filter(resources, matching_source?)
```
Since we are not using `matching_source?` somewhere else, we can just place it inside Enum.find/2
```suggestion
    # Filter resources by language and mode.
    resource_values = Enum.filter(resources, fn resource ->
      resource.language_id == language.id && ResourceValue.supports_mode(resource, mode)
    end)
```

joaothallis replied:
Yes, nice suggestion! I applied here: [refactor(resource): use Enum.find instead of Enum.filter](https://github.com/turnhub/flow_runner/pull/247/commits/b4ce5ef3997d95fcc00060385f0c3e39633add73)

santiagocardo replied:
@joaothallis Using `Enum.find/2` looks way better, thanks 💯
However, this does not solve my original suggestion about the anonymous function 😅 Can we also include that?

santiagocardo reviewed (approved):
LGTM!

@joaothallis Can we please add a description of the issue and how this solves it?

joaothallis commented:
Thank you for review @santiagocardo! I added a description for the PR

joaothallis committed:
refactor(resource): use Enum.find instead of Enum.filter

Since the implementation uses first resource it's not necessary to get
all resources associated with the language and mode

santiagocardo commented on test/resource_test.exs:
```diff
@@ -24,18 +24,29 @@ defmodule ResourceTest do
     assert {:ok, %ResourceValue{value: "french"}} =
              Resource.matching_resource(resource, "fra", "TEXT", @flow)
+
+    # no matching languages for resource should use default language
+    assert {:ok, %ResourceValue{value: "english"}} =
+             Resource.matching_resource(resource, "por", "USSD", @flow)
```
@joaothallis I'm actually not sure about this change. This could generate unexpected behaviors. If the current language is Portuguese, why would that return something that is in English?

Is there any other test case that allows this scenario?

joaothallis replied:
@santiagocardo it's returning English because it's default  ([first language in the list of flow languages](https://github.com/turnhub/flow_runner/pull/247/files#diff-822796d199d7c9b2869fe88753b8ee4acb82709e49282beec40cfb2b1fa2f8a5R10)) language

joaothallis commented:
Since this change changes the current behaviour I'm closing it
